### PR TITLE
Recovery simulations

### PR DESCRIPTION
**Pull request by annehaley**
This PR adds a new Simulation type, "Recovery Scenario". For its input arguments, the user can select the following:

- an existing SimulationResult object from a node failure simulation
- a recovery mode from a list of strings: random, betweenness, degree, information, eigenvector, load, closeness, second order

image

The implementation of this simulation utilizes a function written by Jack Watson at NEU, shared with us for this simulation. The options available in that function determine the list of available recovery modes.

The animation of the results of this simulation only required a slight modification to the `NodeFailureAnimation` component, now renamed `NodeAnimation`. The component now accepts a list of `node_recoveries` alongside the list of `node_failures`. If `node_recoveries` is specified, the animation will subtract the current slice of recovered nodes from the list of failed nodes.

image

### CODE REVIEW

**Comment by AlmightyYakob** [comment]
@annehaley I pushed [`476273f`](https://github.com/OpenGeoscience/uvdat/pull/23/commits/476273f23864859f4140e599eda42bf9366a0008), I hope you don't mind. I noticed that in `construct_edge_list`, `node.adjacent_nodes` was being called once per node, causing a SQL query to be made for every node in the graph. Since the amount of nodes we currently have is only in the hundreds, and we only run this app locally, those _n_  sql queries don't have a humongous impact. However, if either of those two situations were to change, that function would become very slow indeed. I updated that function to instead make 2 SQL queries, which seems to perform much better, and should be able to handle much larger graphs.

I believe I tested it enough to show it has feature parity with the existing implementation, but please give it a try to make sure I'm not missing anything, and/or otherwise lmk what you think.

**Comment by annehaley** [comment]
I tried out [476273f](https://github.com/OpenGeoscience/uvdat/pull/23/commits/476273f23864859f4140e599eda42bf9366a0008) and it works well. Thank you for optimizing that.

File: uvdat/core/tasks/simulations.py [hunk 1]
```diff
@@ -115,21 +188,36 @@ def get_available_simulations(city_id: int):
...
-            }
-            for a in available['args']
-        ]
+            args.append(
+                {
+                    'name': a['name'],
+                    'options': options,
+                }
+            )
+        available['args'] = args
```
[the remark this answers is not in the document]
Comment (annehaley) [reply uvdat/core/tasks/simulations.py:220]: Sure, I agree. Perhaps you could make an issue and record any ideas you'd have for storage alternatives? 

Storage as a database model is usually our default, but I think we would need something more specialized. The simulation types we store represent functions that have been implemented on the server, so they aren't objects that would ever be created by a user, and they have to maintain a pointer to a function. As long as we consider these points, I am open to alternatives.

[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply uvdat/core/tasks/simulations.py:219]: Sure, I've filed #28, so we can track that independently and keep this as is.

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**